packages/web-console/src/scenes/Import/ImportCSVFiles/table-schema-columns.tsx

```diff
@@ -0,0 +1,187 @@
+  { label: "AUTO", value: "" },
+  { label: "BOOLEAN", value: "BOOLEAN" },
+  { label: "BYTE", value: "BYTE" },
+  { label: "DOUBLE", value: "DOUBLE" },
+  { label: "DATE", value: "DATE" },
+  { label: "FLOAT", value: "FLOAT" },
+  { label: "INT", value: "INT" },
+  { label: "LONG", value: "LONG" },
+  { label: "SHORT", value: "SHORT" },
+  { label: "CHAR", value: "CHAR" },
+  { label: "STRING", value: "STRING" },
+  { label: "SYMBOL", value: "SYMBOL" },
+  { label: "TIMESTAMP", value: "TIMESTAMP" },
+]
+
+const Row = styled(Box).attrs({
+  flexDirection: "column",
+  align: "flex-start",
+  gap: "2rem",
+})`
+  width: 100%;
+`
+
+const Columns = styled(Box).attrs({
+  align: "flex-start",
+  justifyContent: "space-around",
+  gap: "1rem",
+})`
+  width: 100%;
+
+  > button {
+    padding-left: 0.5rem;
+    padding-right: 0.5rem;
+  }
+`
+
+export const TableSchemaColumns = ({ schema }: { schema: SchemaColumn[] }) => {
+  const { setValue, watch } = useFormContext()
+  const { fields, append, remove } = useFieldArray({
+    name: "schemaColumns",
+  })
+  const [columnNames, setColumnNames] = useState<string[]>(
+    schema.map((c) => c.name),
+  )
+  const [columnTypes, setColumnTypes] = useState<string[]>(
+    schema.map((c) => c.type),
+  )
+
+  const watchTimestamp = watch("timestamp")
+
+  useEffect(() => {
+    setValue("schemaColumns", schema)
+  }, [schema])
+
+  return (
+    <>
+      {fields.map((field, index) => (
+        <Drawer.GroupItem direction="column" key={field.id}>
+          <Row
+            gap="2rem"
+            flexDirection="column"
+            align="flex-start"
+            key={field.id}
+          >
+            <Columns>
+              <Text color="foreground">{index + 1}</Text>
+              <Box gap="1rem" flexDirection="column">
+                <Box gap="1rem" align="flex-end">
+                  <Form.Item name={`schemaColumns.${index}.name`} label="Name">
+                    <Form.Input
+                      name={`schemaColumns.${index}.name`}
+                      onChange={(e) => {
+                        const cols = [...columnNames]
+                        cols[index] = e.target.value
+                        setColumnNames(cols)
+                      }}
+                    />
+                  </Form.Item>
+                  <Form.Item name={`schemaColumns.${index}.type`} label="Type">
+                    <Form.Select
+                      name={`schemaColumns.${index}.type`}
+                      options={supportedColumnTypes}
+                      onChange={(e) => {
+                        const cols = [...columnTypes]
+                        cols[index] = e.target.value
+                        setColumnTypes(cols)
+                      }}
+                    />
+                  </Form.Item>
+
+                  <IconWithTooltip
+                    icon={
+                      <Button
+                        disabled={columnTypes[index] !== "TIMESTAMP"}
```

Comment:
perhaps instead let's render this button only when `columnTypes[index] === "TIMESTAMP"`?